OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -152,8 +156,14 @@ public void AddSmudge(CPos loc)
 				return;
 
 			if (hasSmoke && Game.CosmeticRandom.Next(0, 100) <= Info.SmokeChance)
+			{
+				var maxOffsetDistance = Info.MaxSmokeOffsetDistance.Length;
+				var x = Game.CosmeticRandom.Next(-maxOffsetDistance, maxOffsetDistance);
+				var y = Game.CosmeticRandom.Next(-maxOffsetDistance, maxOffsetDistance);
```

Comment:
could we perhaps inline these variables and avoid using random if MaxSmokeOffsetDistance is set to 0?